### PR TITLE
cicd: dev builds

### PR DESCRIPTION
**Pull request by johnandersen777**
- Build on each push to dev, archive as tar.gz, release `dev-build.tar.gz` as GitHub release artifact.
  - Example: https://github.com/pdxjohnny/virtualbox-kvm/releases/tag/dev-build
- Build on pull request.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      run: |
+        tar czvf dev-build.tar.gz out/
+        sha384sum dev-build.tar.gz | tee dev-build.tar.gz.sha384sum
+    - uses: ncipollo/release-action@v1
+      if: github.event_name == 'push'
+      with:
+        artifacts: "dev-build.tar.gz,dev-build.tar.gz.sha384sum"
+        allowUpdates: true
+        replacesArtifacts: true
+        body: 'Build for commit ${{ github.sha }}'
```
Comment (johnandersen777) [inline .github/workflows/build.yml]:
```suggestion
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          libc6-dev-i386 lib32stdc++6 libtpms-dev
+    - name: Build
+      run: |
+        ./configure --with-kvm --disable-kmods --disable-docs --disable-hardening --disable-java
+        source ./env.sh
+        kmk -j $(($(nproc) * 4))
+    - name: Archive
+      if: github.event_name == 'push'
+      run: |
+        tar czvf dev-build.tar.gz out/
```
Comment (johnandersen777) [inline .github/workflows/build.yml]:
```suggestion
        echo '${{ github.sha }}' > out/dev-build-commit.txt
        tar czvf dev-build.tar.gz out/
```

File: README.md [hunk 3]
```diff
@@ -26,8 +26,26 @@ workload.
...
 
-There are no prebuilt packages of VirtualBox KVM and it needs to be built from
-source. The process of building VirtualBox from source can be found
+Prebuilt packages for the latest development commit of of VirtualBox KVM can be
+found at https://github.com/cyberus-technology/virtualbox-kvm/releases/tag/dev-build
+
+Optionally download the sha384sums and validate.
+
+```shell
+$ sha384sum -c dev-build.tar.gz.sha384sums
```
Comment (johnandersen777) [inline README.md]:
```suggestion
$ sha384sum -c dev-build.tar.gz.sha384sum
```

**Comment by parthy** [comment]
Thanks for your interest! Our current development model makes it a bit complicated to just accept PRs on GitHub, but if you don't mind, a colleague will reach out to you about this change and what we can do with it.

**Comment by johnandersen777** [comment]
That sounds great! I completely understand, I have interacted with similar models. Please feel free to email me at johnandersenpdx (at) gmail.com

**Pull request closed without merging**